sql: TestEvaluatedMemoryIsChecked failed

sql.TestEvaluatedMemoryIsChecked [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4616576&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4616576&tab=artifacts#/) on release-22.1 @ [f5b42a4e3bbdadb25aee89a41871239a624d538d](https://github.com/cockroachdb/cockroach/commits/f5b42a4e3bbdadb25aee89a41871239a624d538d):


```
=== RUN   TestEvaluatedMemoryIsChecked
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/233e6aa31ff7ad29c896f0479cb5d0c4/logTestEvaluatedMemoryIsChecked2184661387
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvaluatedMemoryIsChecked
    builtin_mem_usage_test.go:123: -- test log scope end --
--- FAIL: TestEvaluatedMemoryIsChecked (31.58s)
=== RUN   TestEvaluatedMemoryIsChecked/#01
    builtin_mem_usage_test.go:119: Expected "SELECT crdb_internal.no_constant_folding(length(repeat('abc', 70000000)))" to OOM, but it didn't
    --- FAIL: TestEvaluatedMemoryIsChecked/#01 (30.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvaluatedMemoryIsChecked.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13976

sql.TestEvaluatedMemoryIsChecked [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4661205&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4661205&tab=artifacts#/) on release-22.1 @ [face5a80eef3bc0e65d31fc411a1c37a18ca9503](https://github.com/cockroachdb/cockroach/commits/face5a80eef3bc0e65d31fc411a1c37a18ca9503):


```
=== RUN   TestEvaluatedMemoryIsChecked
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/233e6aa31ff7ad29c896f0479cb5d0c4/logTestEvaluatedMemoryIsChecked1118147146
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvaluatedMemoryIsChecked
    builtin_mem_usage_test.go:123: -- test log scope end --
--- FAIL: TestEvaluatedMemoryIsChecked (4.69s)
=== RUN   TestEvaluatedMemoryIsChecked/#00
    builtin_mem_usage_test.go:119: Expected "SELECT length(repeat('abc', 70000000))" to OOM, but it didn't
    --- FAIL: TestEvaluatedMemoryIsChecked/#00 (4.15s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvaluatedMemoryIsChecked.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestEvaluatedMemoryIsChecked [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4685974&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4685974&tab=artifacts#/) on release-22.1 @ [7fbd04334b4a5cd369d9df1afdcfa3f49707989d](https://github.com/cockroachdb/cockroach/commits/7fbd04334b4a5cd369d9df1afdcfa3f49707989d):


```
=== RUN   TestEvaluatedMemoryIsChecked
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/233e6aa31ff7ad29c896f0479cb5d0c4/logTestEvaluatedMemoryIsChecked1765935482
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvaluatedMemoryIsChecked
    builtin_mem_usage_test.go:123: -- test log scope end --
--- FAIL: TestEvaluatedMemoryIsChecked (4.30s)
=== RUN   TestEvaluatedMemoryIsChecked/#01
    builtin_mem_usage_test.go:119: Expected "SELECT crdb_internal.no_constant_folding(length(repeat('abc', 70000000)))" to OOM, but it didn't
    --- FAIL: TestEvaluatedMemoryIsChecked/#01 (3.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78486 sql: TestEvaluatedMemoryIsChecked failed [C-test-failure O-robot T-sql-queries branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvaluatedMemoryIsChecked.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/builtin_mem_usage_test.go
+++ b/pkg/sql/builtin_mem_usage_test.go
@@ -29,7 +29,7 @@ import (
 // initial database connection, but small enough to overflow easily. It's set
 // to be comfortably large enough that the server can start up with a bit of
 // extra space to overflow.
-const lowMemoryBudget = 800000
+const lowMemoryBudget = 1 << 20 /* 1MiB */
 
 // rowSize is the length of the string present in each row of the table created
 // by createTableWithLongStrings.

